diff --git a/build_common/SConscript b/build_common/SConscript
index 46a6735..61c5889 100755
--- a/build_common/SConscript
+++ b/build_common/SConscript
@@ -55,6 +55,9 @@ else:
     default_arch = platform.machine()
     if target_os == 'windows':
         default_arch = default_arch.lower()
+    if target_os == 'linux':
+        if default_arch in ['i586', 'i686']:
+         default_arch = 'x86'

 target_arch = ARGUMENTS.get('TARGET_ARCH', default_arch) # target arch

